src/js/api/shipment.js

```diff
@@ -0,0 +1,319 @@
+    options = typeof options === "function" || options === undefined ? {} : options;
+    const url = `${this.url}shipments/${shipment.number}`;
+    options = Object.assign(options, {
+        url: url,
+        method: "PUT",
+        auth: true,
+        dataJ: shipment
+    });
+    options = this._build(options);
+    return this._cacheURL(options.url, options, callback);
+};
+
+/**
+ * Updates a shipment on RIPE Core.
+ *
+ * @param {Object} shipment The shipment to update, with options such as:
+ *  - 'number' - The existing shipment number.
+ *  - 'status' - The shipment status.
+ *  - 'description' - A brief description of this shipment.
+ *  - 'weight' - Weight of the package being shipped.
+ *  - 'weight_units' - Weight units (e.g. "kilograms" or "pounds").
+ *  - 'courier' - The courier for this shipment.
+ *  - 'service' - The transportation service used.
+ *  - 'package' - The package used.
+ *  - 'tracking_number' - The tracking number associated with this shipment.
+ *  - 'tracking_url' - The tracking URL associated with this shipment.
+ *  - 'carbon_neutral' - If this shipment should be carbon neutral.
+ *  - 'access_point_delivery' - If this shipment should be sent to an access point (e.g. "no", "optional" or "mandatory").
+ *  - 'pickup' - Pickup information.
+ *    - 'offset' - Time offset for pickup scheduling relative to shipment creation (millis).
+ *    - 'ready_time' - Opening hours for shipper facility (millis).
+ *    - 'close_time' - Closing hours for shipper facility (millis).
+ *  - 'shipping_date' - The date the shipment ended.
+ *  - 'delivery_date' - The date the shipment began.
+ *  - 'shipper' - Shipper's information.
+ *    - 'name' - Shipper's contact name.
+ *    - 'phone' - Shipper's contact phone.
+ *    - 'address' - Shipper's address.
+ *  - 'customer' - Customer's information.
+ *    - 'name' - Customer's contact name.
+ *    - 'phone' - Customer's contact phone.
+ *    - 'address' - Customer's address.
+ *  - 'orders' - A list of RIPE Core order numbers.
+ *  - 'attachments' - A list of RIPE Core attachment IDs.
+ * @param {Object} options An object of options to configure the request.
+ * @returns {Promise} The shipment's data.
+ */
+ripe.Ripe.prototype.updateShipmentP = function(shipment, options) {
+    return new Promise((resolve, reject) => {
+        this.updateShipment(shipment, options, (result, isValid, request) => {
+            isValid ? resolve(result) : reject(new ripe.RemoteError(request, null, result));
+        });
+    });
+};
+
+/**
+ * Deletes a shipment by number.
+ *
+ * @param {Number} number The number of the shipment to delete.
+ * @param {Object} options An object of options to configure the request.
+ * @param {Function} callback Function with the result of the request.
+ * @returns {XMLHttpRequest} The XMLHttpRequest instance of the API request.
+ */
+ripe.Ripe.prototype.deleteShipment = function(id, options, callback) {

```

Comment:
Since it's by number, perhaps change id to number
```suggestion
ripe.Ripe.prototype.deleteShipment = function(number, options, callback) {

```